src/JestExt.ts

```diff
@@ -599,4 +527,22 @@ export class JestExt {
 
     vscode.debug.startDebugging(vscode.workspace.workspaceFolders[0], configuration)
   }
+
+  onDidCloseTextDocument(document: vscode.TextDocument) {
+    this.removeCachedTestResults(document)
+  }
+
+  onDidSaveTextDocument(editor: vscode.TextEditor, document: vscode.TextDocument) {
+    this.removeCachedTestResults(document)
+    this.triggerUpdateDecorations(editor)
+  }
+
+  removeCachedTestResults(document: vscode.TextDocument) {
+    if (!document || document.isUntitled) {
+      return
+    }
+
+    const filePath = document.fileName
+    this.testResultProvider.removeCachedResults(filePath)
+  }

```

Comment:
nice, breaking up the monolith

Reply:
Thanks. Figured I'd practice some test-driven development. I'm happy we both like how it forces you to add seams to write tests. I'm curious if they'll help integrate changes in the long run or just end up "too fragile".

Reply:
likely the former, people can trust their changes work as expected